### DISCUSSION TITLE
some newbie question want to know

### DISCUSSION THREAD

**Discussion by gen3111620** [milvus-io/milvus#27974] [Q&A and General discussion]
Hi, Thanks the good resource, 

I'd like to know a few things:

1. When I insert 10,000,000 records into the collections and create an index, does the completion of index creation mean that all the data has been indexed? Or do I need to keep checking utility.index_building_progress to confirm if all 10,000,000 rows are indexed?

2. When I insert new data, will it automatically be indexed?

3. Can I perform a search on newly inserted data immediately after insertion?

4. Lastly, I want to inquire if the time it takes to search 10,000,000 records with 512 dimensions using IVF_FLAT and a single record is normal.

the results as below is elapsed time (sec)
<img width="869" alt="圖片" src="https://github.com/milvus-io/milvus/assets/31429532/4a2bc9c8-f9cf-4932-9d39-64af39eb76cf">

Version : Latest milvus standalone (no any setup) just run sudo docker-compose up -d
CPU :  Intel(R) Xeon(R) Gold 6138T CPU @ 2.00GHz
MEM : 64g

Thanks a lot

**Comment by filip-halt** [comment]
1. Calling create_index() will block until all the inserted data has been indexed. Any data inserted after you will need to check the index_build_progress
2. Yes
3. Yes, but if you need read after write you will need a higher consistency level set on the collection (strong)
4. Depends on a lot of things.

**Comment by gen3111620** [comment]
Thanks

regarding question 2, I would like to know the frequency of automatic updates, whether they occur every few minutes or hours, 
and is possible to configure this schedule or manual update indexed?

**Reply by yhmo** [reply]
Milvus manages data by segments, each segment has an independent index.
When you inserting data, the data is accumulated in a "growing" segment, once the segment size reach a threashold(300~500MB), the segment is turned into a "sealed" segment. If you insert new data, another new "growing" segment is created.
Only "sealed" segment can be indexed, the milvus index node can detect new sealed segments and build index for them. No configurations can affect the index scheduler, except a parallel tasks number. No need to change this value.
```
indexNode:
  scheduler:
    buildParallel: 1
```

**Reply by yushangdi** [reply]
Does `buildParallel` mean how many parallel threads are used to build the index? Thanks!

**Reply by yhmo** [reply]
buildParallel means how many **tasks** can be executed at the same time, not about threads.
